### PR TITLE
[Serve][1/n] Serve logging config API

### PR DESCRIPTION
**Pull request by sihanwang41**
## Why are these changes needed?

LoggingConfigSchema will be located in:

- global level: ServeDeploySchema
- application level: ServeApplicationSchema
- deployment level: DeploymentSchema

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/api.py
Comment (sihanwang41) [inline python/ray/serve/api.py:339]: I can remove this change, since i will have follow up change in the next pr.

File: python/ray/serve/config.py
Comment (GeneDer) [inline python/ray/serve/config.py:284]: Just learned those have numeric values XD Should we add an validation on the allowed values?

File: python/ray/serve/schema.py
Comment (GeneDer) [inline python/ray/serve/schema.py:78]: Same for this maybe add an validation

**Review by GeneDer — APPROVED** [review]
LGTM!

File: python/ray/serve/schema.py
Comment (GeneDer) [inline python/ray/serve/schema.py:104]: Kinda after thought, this can be more foolproof by doing `if v not in list(EncodingType)` 🙃

File: python/ray/serve/config.py
Comment (edoakes) [reply python/ray/serve/config.py:284]: Oh jeez it's not even an enum... https://github.com/python/cpython/blob/main/Lib/logging/__init__.py#L98

As this is written, in the config file users would need to look up the levels and convert them to an integer value... can we accept a `Union[None, str, int]` and convert passed strings to ints under the hood?

**Review by edoakes — CHANGES_REQUESTED** [review]

File: python/ray/serve/schema.py
Comment (zcin) [inline python/ray/serve/schema.py:68]: Why is LoggingConfigSchema different from LogginConfig? Is it possible to unify them?

File: python/ray/serve/schema.py
Comment (sihanwang41) [reply python/ray/serve/schema.py]: done!

File: python/ray/serve/schema.py
Comment (sihanwang41) [reply python/ray/serve/schema.py]: done.

File: python/ray/serve/schema.py
Comment (sihanwang41) [reply python/ray/serve/schema.py]: I get rid of the config.py change and only keep the schema.py.

File: python/ray/serve/schema.py
Comment (edoakes) [inline python/ray/serve/schema.py:118]: nit: use `set(list(EncodingType))` for the error message so it doesn't need to be updated if we add a new one

File: python/ray/serve/schema.py
Comment (edoakes) [inline python/ray/serve/schema.py:94]:
```suggestion
            "Log level for the serve logs. Defaults to INFO. You can set it to "
            "'DEBUG' to get more detailed debug logs."
```

File: python/ray/serve/schema.py
Comment (edoakes) [inline python/ray/serve/schema.py:109]:
```suggestion
            "Whether to enable access logs for each request."
```

**Review by edoakes — APPROVED** [review]

**Comment by edoakes** [comment]
Given this isn't hooked up/implemented yet, let's wait until after branch cut to merge it.

**Comment by sihanwang41** [comment]
> Given this isn't hooked up/implemented yet, let's wait until after branch cut to merge it.


> Given this isn't hooked up/implemented yet, let's wait until after branch cut to merge it.

the branch is cut already! https://github.com/ray-project/ray/tree/releases/2.8.0

**Review by zcin — APPROVED** [review]

**Pull request merged**
